src/pterms/Pterm.h

```diff
@@ -125,7 +125,9 @@ class Pterm {
     void     setId(int i)   { id.x = i; }
 
     void     shrink(int s)   { header.size -= s; }
-    void     copyTo(Pterm& to);
+
+    const PTRef* begin() const { return args; }
+    const PTRef* end() const { return args + size(); }
```

Comment:
This is very useful! 
I would just like to add somewhere a big warning that using the range-based for-loop is safe only when no new term can be created in the body of the loop. Otherwise we could be surprised by some unexpected crashes.

Reply:
Yes, that's true, and I suppose we both have been there...  It is true also for the traditional for-loop.  Would you have any ideas how to implement such a warning?  Were you just thinking of a comment, or is there something else that could be done?

Reply:
I meant just a comment, I don't know about anything else that we could do to enforce this.

Reply:
It would be nice to have a feature in c++ where you can declare a variable to be const within a given scope, like you can effectively do for functions.